### ISSUE TITLE
Bug in the `Style/MapToHash` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12429]
## Actual behavior

I have the following code:
```ruby
def my_method
  hsh = { foo: 1, bar: 2 }
  # hsh = nil

  hsh&.map { |k, v| [k.to_s, v.next] }.to_h
end
```

I run rubocop with autocorrection and get:
```ruby
def my_method
  hsh = { foo: 1, bar: 2 }
  # hsh = nil

  hsh&.to_h { |k, v| [k.to_s, v.next] }
end
```

## Expected behavior

```ruby
def my_method
  hsh = { foo: 1, bar: 2 }
  # hsh = nil

  hsh.to_h { |k, v| [k.to_s, v.next] }
end
```

## RuboCop version

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by koic** [comment]
This change will break the compatibility of autocorrection, so I'll close it. Please refer to https://github.com/rubocop/rubocop/pull/12431#discussion_r1412586066 as well

**Comment by ydakuka** [comment]
I disagree. In this example, autocorrection results in different behavior.

**Comment by koic** [comment]
In my tests, the autocorrection made to the presented code appear to be compatible. Removing the safe navigation operator from autocorrected code actually creates incompatibilities.

**Comment by ydakuka** [comment]
Before autocorrection:
```ruby
2.7.6 :002 > hsh = { foo: 1, bar: 2 }
 => {:foo=>1, :bar=>2} 
2.7.6 :003 > hsh&.map { |k, v| [k.to_s, v.next] }.to_h
 => {"foo"=>2, "bar"=>3} 
2.7.6 :004 > hsh = nil
 => nil 
2.7.6 :005 > hsh&.map { |k, v| [k.to_s, v.next] }.to_h
 => {}
```

After autocorrection:
```ruby
2.7.6 :006 > hsh = { foo: 1, bar: 2 }
 => {:foo=>1, :bar=>2} 
2.7.6 :007 > hsh&.to_h { |k, v| [k.to_s, v.next] }
 => {"foo"=>2, "bar"=>3} 
2.7.6 :008 > hsh = nil
 => nil 
2.7.6 :009 > hsh&.to_h { |k, v| [k.to_s, v.next] }
 => nil
```

**Comment by koic** [comment]
Ah, understand. I will work on this again. Thank you!
